addons/mail/static/src/model/model_core.js

```diff
@@ -49,11 +49,18 @@ export function addFields(modelName, fields) {
     }
 }
 
+/**
+ * @typedef {Object} LifecycleHookDefinition
+ * @property {function} [_created] Hook triggered after a model record is created.
+ * @property {function} [_willCreate] Hook triggered when a model record is going to be created.
+ * @property {function} [_willDelete] Hook triggered when a model record is going to be deleted.
```

Comment:
I would still remove the `model` word from here, it just makes reading harder `records` are always related to `models`.